pkgs/by-name/sm/smtp4dev/package.nix

```diff
@@ -0,0 +1,59 @@
+{ lib, buildDotnetModule, fetchFromGitHub, dotnetCorePackages, buildNpmPackage }:
+let
+
+  version = "3.3.4";
+  gsrc = fetchFromGitHub {
+    owner = "rnwood";
+    repo = "smtp4dev";
+    rev = "${version}";
+    hash = "sha256-ARq5OpFJ4o9KdBXvzOx7QLB8GNfmXWjO0RR4jKP8qRI=";
+  };
+
+  webgui = buildNpmPackage rec {
+    pname = "smtp4dev-webgui";
+    version = "3.3.4";
+
+    src = gsrc + "/Rnwood.Smtp4dev/ClientApp";
+
+    npmDepsHash = "sha256-VBcfRKYe/uPf6urWuLI5TrnX9bgiKiZKo+N4zL7O3SM=";
+    postInstall = ''
+      rm -rf $out/*
+      mkdir -p $out
+      cp  -R ../wwwroot $out/wwwroot
+    '';
+  };
+in
+buildDotnetModule rec {
+  pname = "smtp4dev";
+  src = gsrc;
+  inherit version;
+
+  dotnet-sdk = dotnetCorePackages.sdk_8_0;
+  dotnet-runtime = dotnetCorePackages.aspnetcore_8_0;
+
+  projectFile = [ "smtp4dev.sln" ];
+  nugetDeps = ./deps.nix;
+
+  executables = [ "Rnwood.Smtp4dev" ];
+
+  patchPhase = ''
+    substituteInPlace Rnwood.Smtp4dev/Rnwood.Smtp4dev.csproj  --replace '$(SkipClientApp)' 'true'
```

Comment:
In my case I decided to just make the two "Projects", webgui and server separated in the build process. I don't know which one is better, Instead of using the: **envar**? SkipClientApp I replaced it so it doesn't build the webgui on the same step.

I don't know which of the two is preferable, I usually like to separate concerns.


I see that you patchelf the binary, is that necessary, should we include that here?

Feel free to add more to the commit, will add you as coauthor.

Reply:
I've updated this PR, the last commit adds you as a coauthor. I like it more the way you did it regarding the webgui in the end.